packages/components/src/components/FileUploader/FileUploader.tsx

```diff
@@ -0,0 +1,282 @@
+  icon: keyof typeof HeroOutlineIcons;
+} => {
+  switch (status) {
+    case "success": {
+      return { icon: "PaperClipIcon", color: "colorTextSuccess" };
+    }
+    case "error": {
+      return { icon: "XCircleIcon", color: "colorTextError" };
+    }
+    default: {
+      return { icon: "CloudArrowUpIcon", color: "colorTextLink" };
+    }
+  }
+};
+
+const getStatus = (
+  progress: number,
+  fileUrl?: string,
+  errorMessage?: string
+): FileUploaderStatus => {
+  if (progress > 0 && progress < 100) return "loading";
+  if (!isEmpty(fileUrl)) return "success";
+  if (!isEmpty(errorMessage)) return "error";
+  return "waiting";
+};
+
+const CancelUploadButton = (props: Partial<ButtonProps>): ReactElement => {
+  const isMobile = useDown("md");
+
+  if (isMobile) {
+    return (
+      <Button {...props} variant="secondary">
+        Cancel upload
+      </Button>
+    );
+  }
+
+  return (
+    <Button
+      {...props}
+      aria-label="Cancel upload"
+      iconOnly
+      trailingIcon="XMarkIcon"
+      variant="ghost"
+    />
+  );
+};
+
+const RemoveButton = (props: Partial<ButtonProps>): ReactElement => {
+  const isMobile = useDown("md");
+
+  if (isMobile) {
+    return (
+      <Button {...props} aria-label="Remove file" variant="secondary">
+        Remove
+      </Button>
+    );
+  }
+
+  return (
+    <Button
+      aria-label="Remove file"
+      iconOnly
+      trailingIcon="TrashIcon"
+      variant="ghost"
+    />
+  );
+};
+
+const getFileTitle = (label: string, fileUrl?: string): React.ReactElement => {
+  if (fileUrl) {
+    return (
+      <Anchor href={fileUrl} target="_blank">
+        {label}
+      </Anchor>
+    );
+  }
+  return <Text.span color="colorTextStrongest">{label}</Text.span>;
+};
+
+const getFileDescription = (
+  status: FileUploaderStatus,
+  { fileName, fileSize, maxFileSize }: Partial<FileUploaderProps>
+): ReactNode => {
+  const wrapTextInComponent = (value: string) => (
+    <Text.span fontSize="fontSize10">{value}</Text.span>
+  );
+
+  if (status === "loading") {
+    return fileName || "";
+  }
+
+  const name = fileName || (status !== "success" && "No file uploaded");
+  const size =
+    fileSize || (maxFileSize && `Max. file size ${maxFileSize}`) || null;
+
+  if (name && size) {
+    return wrapTextInComponent(`${name} • ${size}`);
+  }
+
+  if (name) {
+    return wrapTextInComponent(name);
+  }
+
+  return null;
+};
+
+/** Visual component to display status of a file upload */
+const FileUploader = React.forwardRef<HTMLDivElement, FileUploaderProps>(
+  (
+    {
+      fileName,
+      fileUrl,
+      fileSize,
+      maxFileSize,
+      errorMessage,
+      label,
+      children,
+      progress = 0,
+      onCancel,
+      onRemove,
+    },
+    ref
+  ) => {
+    const isMobile = useDown("md");
+    const status = getStatus(progress, fileUrl, errorMessage);
+
+    return (
+      <Box.div display="flex" flexDirection="column" gap="space25">
```

Comment:
Can these Box elements be made separate components like `FileUploaderHeader` or `FileUploaderContent`. Something like that?